ocp_resources/data_source.py

```diff
@@ -1,46 +1,38 @@
+from warnings import warn
 from kubernetes.dynamic.exceptions import ResourceNotFoundError
-
-from ocp_resources.constants import TIMEOUT_4MINUTES
 from ocp_resources.persistent_volume_claim import PersistentVolumeClaim
 from ocp_resources.resource import NamespacedResource
+from ocp_resources.volume_snapshot import VolumeSnapshot
 
 
 class DataSource(NamespacedResource):
+    """
+    DataSource object.
+
+    https://kubevirt.io/cdi-api-reference/main/definitions.html#_v1beta1_datasource
+    """
+
     api_group = NamespacedResource.ApiGroup.CDI_KUBEVIRT_IO
 
-    def __init__(
-        self,
-        name=None,
-        namespace=None,
-        client=None,
-        source=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
-        **kwargs,
-    ):
-        super().__init__(
-            name=name,
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
-        self.source = source
+    def __init__(self, source=None, **kwargs):
+        """
+        Args:
+            source (dict): The source of the data, can reference 'pvc' or 'snapshot'.
```

Comment:
do not define here what the api currently supports; we should not need to updae the code in the docstring once the api will support new source types